lib/acpi/acpi_shell.c

```diff
@@ -220,19 +226,93 @@ static int dump_prt(const struct shell *sh, size_t argc, char **argv)
 static int enum_dev(const struct shell *sh, size_t argc, char **argv)
 {
 	struct acpi_dev *dev;
+	ACPI_RESOURCE *res_lst;
 
-	if (argc < 2) {
+	if (argc < 3) {
+		shell_error(sh, "Invalid arguments [Eg: acpi enum PNP0103 0]");
 		return -EINVAL;
 	}
 
-	dev = acpi_device_get(argv[1], 0);
+	dev = acpi_device_get(argv[1], argv[2]);
 	if (!dev || !dev->res_lst) {
 		shell_error(sh, "acpi get device failed for HID: %s", argv[1]);
 		return -EIO;
 	}
 
 	shell_print(sh, "Name: %s", dev->path ? dev->path : "None");
-	dump_dev_res(sh, dev->res_lst);
+
+	if (dev->path) {
+		if (!acpi_current_resource_get(dev->path, &res_lst)) {
+			dump_dev_res(sh, res_lst);
+			acpi_current_resource_free(res_lst);
+		}
+	}
+
+	return 0;
+}
+
+static int enum_all_dev(const struct shell *sh, size_t argc, char **argv)
+{
+	struct acpi_dev *dev;
+
+	for (int i = 0; i < 1000; i++) {
```

Comment:
The magic `1000` seems a bit arbitrary. Where's that coming from? It should probably at least be some `ACPI_MAX_DEV` define.